dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/acl/TrackedEntityInstanceAclReadTests.java

```diff
@@ -0,0 +1,539 @@
+package org.hisp.dhis.tracker.acl;
+
+import com.google.common.collect.Lists;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.apache.commons.collections.ListUtils;
+import org.hisp.dhis.ApiTest;
+import org.hisp.dhis.actions.LoginActions;
+import org.hisp.dhis.actions.RestApiActions;
+import org.hisp.dhis.actions.UserActions;
+import org.hisp.dhis.actions.metadata.MetadataActions;
+import org.hisp.dhis.actions.tracker.EventActions;
+import org.hisp.dhis.actions.tracker.TEIActions;
+import org.hisp.dhis.dto.ApiResponse;
+import org.hisp.dhis.helpers.QueryParamsBuilder;
+import org.hisp.dhis.helpers.file.FileReaderUtils;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.ThrowingSupplier;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import sun.rmi.runtime.Log;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author Stian Sandvold
+ */
+public class TrackedEntityInstanceAclReadTests
+    extends ApiTest
+{
+    private static final String _TEIS = "trackedEntityInstances";
+
+    private static final String _ENROLLMENTS = "enrollments";
+
+    private static final String _EVENTS = "events";
+
+    private static final String _TET = "trackedEntityType";
+
+    private static final String _PROGRAM = "program";
+
+    private static final String _PROGRAMSTAGE = "programStage";
+
+    private static final String _OU = "orgUnit";
+
+    private static final String _TEI = "trackedEntityInstance";
+
+    private static final String _ENROLLMENT = "enrollment";
+
+    private static final String _PROGRAMOWNERS = "programOwners";
+
+    private static final String _OWNEROU = "ownerOrgUnit";
+
+    private static final String _DELETED = "deleted";
+
+    private static final String _ATTRIBUTES = "attributes";
+
+    private static final String _ATTRIBUTE = "attribute";
+
+    private MetadataActions metadataActions;
+
+    private UserActions userActions;
+
+    private TEIActions teiActions;
+
+    private static final List<User> users = new ArrayList<>();
+
+    private class User
```

Comment:
I think this should go to its own class, somewhere in org.hisp.dhis.dto package

Reply:
Now that I read everything, I guess this is more tailored to this specific test, so scratch that. We could reuse it with some modifications, read more here: https://github.com/rest-assured/rest-assured/wiki/Usage#object-mapping.

Reply:
I rewrote and moved the /me response this way. I was about to do the same for /metadata, but after I got into it, I realized it would results in a lot of different files only support this single use-case, so I decided to keep it the way it is. I also moved the User class into it's own file to keep the test itself a bit cleaner :)